client/iface/wgproxy/udp/proxy.go

```diff
@@ -175,14 +228,20 @@ func (p *WGUDPProxy) proxyToLocal(ctx context.Context) {
 			return
 		}
 
-		p.pausedMu.Lock()
-		if p.paused {
-			p.pausedMu.Unlock()
-			continue
+		for {
+			p.pausedCond.L.Lock()
+			if p.paused {
+				p.pausedCond.Wait()
+				if !p.paused {
+					break
+				}
+				p.pausedCond.L.Unlock()
+				continue
+			}
+			break
```

Comment:
Would this be the same but much simpler?
```suggestion
		p.pausedCond.L.Lock()
		for p.paused {
			p.pausedCond.Wait()
```